Backpack/Rating/Classes/BPKRating.swift

```diff
@@ -0,0 +1,255 @@
+    }()
+
+    private var titleViewBottomConstraint: NSLayoutConstraint?
+
+    // MARK: - Init
+    public init(
+        accessibilityLabel: String,
+        title: String,
+        value: Float,
+        ratingScale: BPKRatingScale = .zeroToFive,
+        size: BPKRatingSize = .default,
+        subtitle: String? = nil,
+        showScale: Bool = true
+    ) {
+        self.value = value
+        self.ratingScale = ratingScale
+        self.size = size
+        self.showScale = showScale
+        self.titleView = titleLabel
+        super.init(frame: .zero)
+
+        self.title = title
+        self.subtitle = subtitle
+
+        self.isAccessibilityElement = true
+        self.accessibilityLabel = accessibilityLabel
+
+        setup()
+        updateLookAndFeel()
+    }
+
+    public init(
+        accessibilityLabel: String,
+        value: Float,
+        ratingScale: BPKRatingScale = .zeroToFive,
+        size: BPKRatingSize = .default,
+        subtitle: String? = nil,
+        showScale: Bool = true,
+        titleView: UIView
+    ) {
+        self.value = value
+        self.ratingScale = ratingScale
+        self.size = size
+        self.showScale = showScale
+        self.titleView = titleView
+
+        super.init(frame: .zero)
+
+        self.subtitle = subtitle
+        self.isAccessibilityElement = true
+        self.accessibilityLabel = accessibilityLabel
+
+        setup()
+        updateLookAndFeel()
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("NSCoding not supported")
+    }
+
+    private func setup() {
+        addSubview(horizontalStackView)
+        
+        [
+            ratingValueAndScaleStackView,
+            titleSubtitleStackView
+        ].forEach(horizontalStackView.addArrangedSubview(_:))
+        [
+            ratingValueLabel,
+            ratingScaleLabel
+        ].forEach(ratingValueAndScaleStackView.addArrangedSubview(_:))
+        [titleView, subtitleLabel].forEach(titleSubtitleStackView.addArrangedSubview(_:))
+
+        setupConstraints()
+    }
+
+    private func setupConstraints() {
+        horizontalStackView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            horizontalStackView.topAnchor.constraint(equalTo: topAnchor),
+            horizontalStackView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            horizontalStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            horizontalStackView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+        setupTitleViewConstraint()
+    }
+
+    private func setupTitleViewConstraint() {
+        titleViewBottomConstraint = titleView.bottomAnchor.constraint(
+            lessThanOrEqualTo: ratingValueLabel.lastBaselineAnchor
+        )
+        titleViewBottomConstraint?.priority = .required
+    }
+
+    private func updateLookAndFeel() {
+        subtitleLabel.isHidden = subtitleLabel.text == nil
```

Comment:
Is this needed? nothing will be shown if the text is already nil

Reply:
That is a good idea. I am trying it now

Reply:
I remove this line in this commit.
b793852cf3d59ab5924fdacf661b68cb0ce0b888